### PR TITLE
Replace Python getter functions with properties in sync API

### PR DESCRIPTION
**Pull request by pentschev**
To make `libucxx.pyx` more consistent and Pythonic, replace various getter functions with `@property`.

### CODE REVIEW

File: python/ucxx/_lib/libucxx.pyx [hunk 1]
```diff
@@ -527,9 +548,27 @@ cdef class UCXWorker():
...
+
+    @property
+    def enable_python_future(self):
+        return self._enable_python_future
+
+    def get_address(self):
+        warnings.warn(
+            "UCXWorker.get_address() is deprecated and will soon be removed, "
+            "use the UCXWorker.address property instead",
+            DeprecationWarning,
```
Comment (wence-) [inline python/ucxx/_lib/libucxx.pyx:567]: Just to note that if we want these to be visible to the user, they should be `FutureWarning` not `DeprecationWarning` (see https://docs.rapids.ai/api/cudf/stable/developer_guide/contributing_guide/#deprecating-and-removing-code and https://docs.python.org/3/library/warnings.html#warning-categories)

**Review by wence- — APPROVED** [review]
Minor suggestion to change `DeprecationWarning` to `FutureWarning`, otherwise all great AFAICT!

File: python/ucxx/_lib/libucxx.pyx (hunk 1 above)
Comment (pentschev) [reply python/ucxx/_lib/libucxx.pyx]: Good point, fixed in https://github.com/rapidsai/ucxx/pull/158/commits/62d00b4829d96bea4542a0de8cc8934072c2ee6b

**Review by charlesbluca — APPROVED** [review]

**Comment by pentschev** [comment]
Thanks @charlesbluca and @wence- !

**Pull request merged**